src/actions/business-actions.js

```diff
@@ -1,60 +1,77 @@
 
+function changeLocation(location) {
+    return {
+        type: "CHANGE_LOCATION",
+        location: location
+    }
+}
+
 function selectBusiness(business) {
     return {
-      type: "SELECT_CURRENT_BUSINESS",
-      business: business
+        type: "SELECT_BUSINESS",
+        business: business
     }
 }
 function deselectBusiness(business) {
     return {
-      type: "DESELECT_CURRENT_BUSINESS",
-      business: business
+        type: "DESELECT_BUSINESS",
+        business: business
     }
 }
 
 function fetchBusinessList() {
     return {
-      type: "FETCH_BUSINESS_LIST",
+        type: "FETCH_BUSINESS_LIST",
     }
 }
 
 function fetchBusinessListSuccess(businesses) {
-    
+
     return {
-      type: "FETCH_BUSINESS_LIST_SUCCESS",
-      businesses: businesses
+        type: "FETCH_BUSINESS_LIST_SUCCESS",
+        businesses: businesses
     }
 }
 
 function fetchBusinessListFailure(error) {
     return {
-      type: "FETCH_BUSINESS_LIST_FAILURE",
-      error: error
+        type: "FETCH_BUSINESS_LIST_FAILURE",
+        error: error
     }
 }
 
-function fetchBusinessListAsync(){
-   
-    return function(dispatch){
+function fetchBusinessListAsync(locationStr) {
+
+    return function (dispatch) {
+        console.log("Acttions " + locationStr);
         dispatch(fetchBusinessList())
-        fetch('api/businesses', {
-            method: 'get'
-        }).then(function(response) {
-      
-            response.json().then(actualData => {
-                dispatch(fetchBusinessListSuccess(actualData));
-            }).catch(jsonParsingError => {
-                dispatch(fetchBusinessListFailure(jsonParsingError));    
+        if (locationStr == "") {
+            console.log("We are here")
+            //           locationStr = "berlin"
+            dispatch(fetchBusinessListSuccess([]));
+        } else {
+            console.log("We are here1")
+            // fetch('api/businesses', {
+            fetch(`/api/businesses?location=${locationStr}`, {
```

Comment:
I see what you mean. I thought about it as well, but came to conclusion that it would be better to keep it simple. A business will enter a city and country during registration. This should help to avoid confusion on the client's side (e.g. Paris, France and Paris, US). Considering that a client won't usually know a zip-code of the foreign city, I think we can just give them all businesses with matching location. A client can sort them out manually. Again, it's an MVP, if this app will grow, we will add more stuff (like automatic location detection).